### ISSUE TITLE
Provide a method to retrieve a node by name on graph

### ISSUE THREAD

**Issue by masseelch** [ent/ent#1320]
- [x] I have searched the [issues](https://github.com/facebook/ent/issues) of this repository and believe that this is not a duplicate.

## Summary 💡

Hi again :smile: 

I tend to have my `schema` folder out of the `ent` folder (where the generated code lands). This enabled me to remove the `ent` folder and regenerate the code (for example after removing an a schema).

Therefore I do a lot of my work on external templates, since regenerating will put my extensions back to place after getting rid of the `ent` folder. I have a lot of methods on the generated entities, for example like mentioned [here](https://github.com/ent/ent/issues/1072#issue-772860771).

To generate the code for just one entity, I have to guess the receiver name or, if I want to use the receiver-name provided by ent, loop over all nodes in the schema and skip the models I am not interested in. 

Basically I'd like have this implemented on `Graph` struct: 

```go
func (g Graph) GetTypeByName(n string) *Type {
	for _, n := range g.Nodes {
		if n.Name == n {
			return n
		}
	}
	
	return nil
}
```

Of course the name is just a suggestion :wink: 

There might even be a solution to this I am not yet aware of. In that case I kindly ask to guide me to it. :relaxed:

**Comment by rotemtam** [comment]
This would be useful, I wrote a helper method inside `contrib/entproto` for this, and I guess it would be generally useful for any tooling that's doing intelligent code generation. 

https://github.com/ent/contrib/blob/95e31056bf41adc8ac8e20cf6747bfc621fb3734/entproto/adapter.go#L539-L546

**Comment by a8m** [comment]
Hey @masseelch,
There are 2 ways to implement this logic without adding it to the core library. 

1\. Create a helper function like this and pass it to your template:

```go
// byName returns a node in the graph by its label/name.
func byName(g *gen.Graph, name string) (*gen.Type, error) {
	for _, n := range g.Nodes {
		if n.Name == name {
			return n, nil
		}
	}
	return nil, fmt.Errorf("node %q was not found in the graph", name)
}
```
I added it to our examples. Please see https://github.com/ent/ent/pull/1326 and let me know if it helps or not.

2\. The second option is to write this logic on your template if it's a one-time case:

```gotemplate
{{ $usr := index $.Nodes 0 }}
{{ range $n := slice $.Nodes 1 }}{{ if eq $n.Name "User" }}{{ $usr = $n }}{{ end }}{{ end }}
{{ if ne $usr.Name "User" }}{{ fail "User was not found" }}{{ end }}

{{ with $usr }}
	{{/* Logic goes here */}}
{{ end }}
```
